### PR TITLE
chore: restructure uat module

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
This change is made to include `custom-components` as a separate module inside parent `uat` module in the later PRs.
The previous `uat` module is renamed as t`esting-features` (similar to what is used in aws-greengrass-testing). 

**Why is this change necessary:**
Without this change, we will have to build the custom-components module separately to include its artifacts in the uat module. But with this change, customers and the github workflow can still be run with the same command but with minor changes. 

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [ ] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/testing-features/pom.xml [hunk 1]
```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Nelsonochoam) [inline uat/testing-features/pom.xml:1]: Wouldnt you need a parent pom if you want to do the same as otf? 

I want to have some sort of consistency between our repos (so if we go ahead with this we would also want to update log manager) - What custom components are we placing? I guess tu simulate clients updating the shadow?

Comment (saranyailla) [reply uat/testing-features/pom.xml:1]: `uat` module is this PR is the parent packaging as a pom without any artifacts. Almost all shadow manager UATs need custom components that perform IPC operations.

**Review by MikeDombo — APPROVED** [review]

**Review by Nelsonochoam — APPROVED** [review]

File: uat/testing-features/pom.xml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+<!--
+
+    Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+    SPDX-License-Identifier: Apache-2.0
+
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (Nelsonochoam) [inline uat/testing-features/pom.xml:10]:
```suggestion
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <parent>
        <artifactId>shadow-manager-uat</artifactId>
        <groupId>com.aws.greengrass</groupId>
        <version>1.0-SNAPSHOT</version>
    </parent>

File: uat/testing-features/pom.xml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.aws.greengrass</groupId>
+    <artifactId>shadow-manager-testing-features</artifactId>
```
Comment (Nelsonochoam) [inline uat/testing-features/pom.xml:13]:
```suggestion
    <artifactId>shadow-manager-uat-features</artifactId>
```

File: uat/testing-features/pom.xml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    </properties>
+    <repositories>
+        <repository>
+            <id>greengrass-common</id>
+            <name>greengrass common</name>
+            <!-- CloudFront url fronting the aws-greengrass-testing-standalone in S3-->
+            <url>https://d2jrmugq4soldf.cloudfront.net/snapshots</url>
+        </repository>
+    </repositories>
+    <dependencies>
```
Comment (Nelsonochoam) [inline uat/testing-features/pom.xml:29]: Whatever dependencies are going to be common between the modules should be moved to the parent POM

**Pull request merged**